Apps/Laboratory/test/unit/Services/LaboratoryServiceTests.cs

```diff
@@ -46,44 +46,54 @@ public class LaboratoryServiceTests
         /// The value indicates whether the covid19 test result data source can be accessed or
         /// not.
         /// </param>
+        /// <param name="accessTokenExists">bool value indicating whether access token exists or not.</param>
+        /// <param name="refreshInProgress">bool value indicating whether there is a refresh in progress or not.</param>
         /// <returns>awaitable task.</returns>
         [Theory]
-        [InlineData(ResultType.Success, true)]
-        [InlineData(ResultType.Success, false)]
-        [InlineData(ResultType.Error, true)]
-        public async Task ShouldGetCovid19Orders(ResultType expectedResultType, bool canAccessDataSource)
+        [InlineData(ResultType.Success, true, true, false)]
+        [InlineData(ResultType.Success, false, true, false)]
+        [InlineData(ResultType.Error, true, true, false)]
+        [InlineData(ResultType.Error, true, false, false)]
+        [InlineData(ResultType.ActionRequired, true, true, true)]
+        public async Task ShouldGetCovid19Orders(ResultType expectedResultType, bool canAccessDataSource, bool accessTokenExists, bool refreshInProgress)
         {
-            List<PhsaCovid19Order> covid19Orders = new()
-            {
+            List<PhsaCovid19Order> covid19Orders =
+            [
                 new PhsaCovid19Order
                 {
                     Id = Guid.NewGuid(),
                     Location = "Vancouver",
-                    Phn = "001",
+                    Phn = "0000000001",
                     MessageDateTime = DateTime.Now,
                     MessageId = MockedMessageId + "1",
                     ReportAvailable = true,
                 },
+
                 new PhsaCovid19Order
                 {
                     Id = Guid.NewGuid(),
                     Location = "Vancouver",
-                    Phn = "002",
+                    Phn = "0000000002",
                     MessageDateTime = DateTime.Now,
                     MessageId = MockedMessageId + "2",
                     ReportAvailable = false,
                 },
-            };
+            ];
 
             RequestResult<PhsaResult<List<PhsaCovid19Order>>> delegateResult = new()
             {
                 ResultStatus = expectedResultType,
                 PageSize = 100,
                 PageIndex = 1,
-                ResourcePayload = new() { Result = covid19Orders },
+                ResourcePayload = new()
+                {
+                    Result = covid19Orders, LoadState = new PhsaLoadState
+                        { RefreshInProgress = refreshInProgress },
```

Comment:
Nit: code formatting is off.

```suggestion
                    Result = covid19Orders,
                    LoadState = new PhsaLoadState { RefreshInProgress = refreshInProgress },
```